x/iscn/keeper/keeper.go

```diff
@@ -250,25 +251,19 @@ func (k Keeper) IterateIscnIds(ctx sdk.Context, f func(iscnId IscnId, contentIdR
 	})
 }
 
-func (k Keeper) DeductFeeForIscn(ctx sdk.Context, feePayer sdk.AccAddress, data []byte) error {
-	acc := k.accountKeeper.GetAccount(ctx, feePayer)
-	if acc == nil {
-		return fmt.Errorf("cannot find account %s for deducting fee", feePayer.String())
-	}
+func (k Keeper) DeductFeeForIscn(ctx sdk.Context, msgSender sdk.AccAddress, data []byte, msg sdk.Msg) error {
 	feePerByte := k.GetParams(ctx).FeePerByte
 	feeAmount := feePerByte.Amount.MulInt64(int64(len(data)))
 	fees := sdk.NewCoins(sdk.NewCoin(feePerByte.Denom, feeAmount.Ceil().RoundInt()))
-	if !fees.IsZero() {
-		err := ante.DeductFees(k.bankKeeper, ctx, acc, fees)
-		if err != nil {
-			return err
-		}
-	}
-	return nil
+	return likefeegrant.DeductFeePerByte(
+		ctx,
+		k.accountKeeper, k.bankKeeper, k.feegrantKeeper,
+		msgSender, fees, msg,
+	)
 }
```

Comment:
Seems `iscn.DeductFeeForIscn()` and `likenft.DeductFeePerByte()` have similar logic and both rely on `likefeegrant.DeductFeePerByte()`. 
Is it preferred to refactor them into `likefeegrant.DeductFeePerByte()`?
Though I think this would be really optional.

Reply:
Not quite possible since the `feePerByte := k.GetParams(ctx).FeePerByte` parts on 2 modules are actually getting parameters from different keepers with different parameter types